### PR TITLE
Add xo to coding style

### PR DESCRIPTION
**Pull request by RobLoach**
Another possibility is to stick this logic into `test-transformer`, maybe.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,9 +1,9 @@
 root = true
 
 [*]
-indent_style = space
+indent_style = tab
 charset = utf-8
 end_of_line = lf
-indent_size = 2
+indent_size = 4
```
Comment (RobLoach) [inline .editorconfig]: Not happy with the tab design decision. I believe the majority of them using 2 line spacing currently. Use `--space`

Comment (calebeby) [inline .editorconfig]: Can I change tabs to spaces, or would you rather that I didn't change this PR?

Comment (TimothyGu) [inline .editorconfig]: Can you change it to just like before?

File: index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-exports.name = 'foo';
-exports.inputFormats = ['foo', 'foobar'];
-exports.outputFormat = 'html';
+exports.name = 'foo'
+exports.inputFormats = ['foo', 'foobar']
+exports.outputFormat = 'html'
 
-exports.render = function (str, options) {
-  return str;
+exports.render = function (str) {
```
Comment (calebeby) [inline index.js]: Why was `options` removed? :confused:

File: package.json [hunk 3]
```diff
@@ -9,11 +9,13 @@
...
   "devDependencies": {
-    "test-jstransformer": "^1.0.0"
+    "test-jstransformer": "^1.0.0",
+    "xo": "^0.15.1"
   },
   "scripts": {
     "coverage": "test-jstransformer coverage",
-    "test": "test-jstransformer"
+    "test": "test-jstransformer",
+    "posttest": "xo --no-semicolon index.js"
```
Comment (calebeby) [inline package.json]: I think this is better than putting xo in `test-jstransformer`.

:+1:

File: index.js (hunk 2 above)
Comment (calebeby) [inline index.js]: Oh, I see now. XO sees it isn't used.

**Comment by calebeby** [comment]
Any final objections, or is this good to go?

**Comment by TimothyGu** [comment]
Actually, is there a reason why semicolons are removed?

**Comment by calebeby** [comment]
They aren't necessary, and I think it's prettier without them. That's just my opinion; you can ask @RobLoach since he was the one who removed them.

**Comment by TimothyGu** [comment]
I'm fine with both, but I am under the impression that most repos under this organization use semicolons and it might be more work to switch the style now.

**Comment by calebeby** [comment]
Yeah, it will take a while, but I'm OK with that.

**Comment by RobLoach** [comment]
An alternative would be to bring it directly into `test-jstransformer`. When you run test-jstransformer, it runs xo.

**Comment by calebeby** [comment]
@RobLoach I think I like this better :+1:

**Pull request merged**